devicetypes/smartthings/aeon-led-bulb-6.src/aeon-led-bulb-6.groovy

```diff
@@ -0,0 +1,335 @@
+		result << createEvent(name: "color", value: hexColor)
+		// Send the color as hue and saturation
+		def hsv = rgbToHSV(*colors)
+		result << createEvent(name: "hue", value: hsv.hue)
+		result << createEvent(name: "saturation", value: hsv.saturation)
+		// Reset the values
+		RGB_NAMES.collect { state.colorReceived[it] = null}
+	}
+	// Check if we got all the color temperature values
+	if (WHITE_NAMES.every { state.colorReceived[it] != null}) {
+		def warmWhite = state.colorReceived[WARM_WHITE]
+		def coldWhite = state.colorReceived[COLD_WHITE]
+		log.debug "warmWhite: $warmWhite, coldWhite: $coldWhite"
+		if (warmWhite == 0 && coldWhite == 0) {
+			result = createEvent(name: "colorTemperature", value: COLOR_TEMP_MIN)
+		} else {
+			def parameterNumber = warmWhite ? WARM_WHITE_CONFIG : COLD_WHITE_CONFIG
+			result << response(command(zwave.configurationV2.configurationGet([parameterNumber: parameterNumber])))
+		}
+		// Reset the values
+		WHITE_NAMES.collect { state.colorReceived[it] = null }
+	}
+	result
+}
+
+private dimmerEvents(physicalgraph.zwave.Command cmd) {
+	def value = (cmd.value ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value, descriptionText: "$device.displayName was turned $value")]
+	if (cmd.value) {
+		result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value , unit: "%")
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	log.debug "got ConfigurationReport: $cmd"
+	def result = null
+	if (cmd.parameterNumber == WARM_WHITE_CONFIG || cmd.parameterNumber == COLD_WHITE_CONFIG)
+		result = createEvent(name: "colorTemperature", value: cmd.scaledConfigurationValue)
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	def linkText = device.label ?: device.name
+	[linkText: linkText, descriptionText: "$linkText: $cmd", displayed: false]
+}
+
+def buildOffOnEvent(cmd){
+	[zwave.basicV1.basicSet(value: cmd), zwave.switchMultilevelV3.switchMultilevelGet()]
+}
+
+def on() {
+	commands(buildOffOnEvent(0xFF), 5000)
+}
+
+def off() {
+	commands(buildOffOnEvent(0x00), 5000)
+}
+
+def refresh() {
+	commands([zwave.switchMultilevelV3.switchMultilevelGet()] + queryAllColors())
+}
+
+def ping() {
+	log.debug "ping().."
+	unschedule(offlinePing)
+	runEvery5Minutes(offlinePing)
+	command(zwave.switchMultilevelV3.switchMultilevelGet())
+}
+
+def offlinePing() {
+	log.debug "offlinePing()..."
+	sendHubCommand(new physicalgraph.device.HubAction(command(zwave.switchMultilevelV3.switchMultilevelGet())))
+}
+
+def setLevel(level) {
+	setLevel(level, 1)
+}
+
+def setLevel(level, duration) {
```

Comment:
I think your old code was right, I just got confused by the units.

Reply:
I think I'll stick with this current change since this bulb sets the dimming duration using a configuration parameter anyway. Dimming takes a while on this bulb, and 5 seconds seems to be the right amount of time to get responses back after it completes.